src/rpc/blockchain.cpp

```diff
@@ -2450,6 +2451,15 @@ static RPCHelpMan getblockstats()
             }
 
             CAmount txfee = tx_total_in - tx_total_out;
+
+            if (tx->nType == TRANSACTION_ASSET_UNLOCK) {
+                if (const auto opt_assetUnlockTx = GetTxPayload<CAssetUnlockPayload>(*tx)) {
+                    txfee = opt_assetUnlockTx->getFee();
+                } else {
+                    txfee = 0;
+                }
```

Comment:
```suggestion
                const auto opt_assetUnlockTx = GetTxPayload<CAssetUnlockPayload>(*tx);
                CHECK_NONFATAL(opt_assetUnlockTx.has_value());
                txfee = opt_assetUnlockTx->getFee();
```

Reply:
Taking a step further... How about bc3a811730?

Reply:
Looks reasonable to me!

Reply:
```
    if (calculate_fee) {
        CAmount fee = amt_total_in - amt_total_out;
        if (tx.nType == TRANSACTION_ASSET_UNLOCK) {
            fee = CHECK_NONFATAL(GetTxPayload<CAssetUnlockPayload>(tx))->getFee();
        }
```

Need to call `GetTxPayload` twice, I guess over-head is not huge?

Reply:
That's just reading/deserializing from datastream, no heavy ops. Should be fine imo.